spring-cloud-dataflow-docs/src/main/asciidoc/appendix-migration-guide.adoc

```diff
@@ -0,0 +1,381 @@
+All the out-of-the-box applications are pre-built for both Kafka and Rabbit and they’re 
+readily available for use as maven artifacts [link:http://repo.spring.io/libs-snapshot/org/springframework/cloud/stream/app/[stream] / link:http://repo.spring.io/libs-snapshot/org/springframework/cloud/task/app/[task]] or docker images [link:https://hub.docker.com/r/springcloudstream/[stream] / link:https://hub.docker.com/r/springcloudtask/[task]]
+Changing the binder requires selecting the right binder link:https://github.com/spring-cloud/spring-cloud-stream/blob/master/spring-cloud-stream-docs%2Fsrc%2Fmain%2Fasciidoc%2Fspring-cloud-stream-overview.adoc#binder-selection[dependency]. Alternatively, you can download the pre-built application from this version of link:http://start-scs.cfapps.io/[Spring Initializr] with the desired “binder-starter” dependency
+
+==== Named Channels
+
+Fundamentally, all the messaging channels are backed by pub/sub semantics. Unlike Spring XD, the 
+messaging channels are backed only by `topics` or `topic-exchange` and there’s no representation of 
+`queues` in the new architecture.
+
+* `${xd.module.index}` is not supported anymore; instead, you can directly interact with named 
+destinations
+* `stream.index` changes to `:<stream-name>.<label/app-name>`
+** _for instance:_ `ticktock.0` changes to `:ticktock.time`
+* “topic/queue” prefixes are not required to interact with named-channels
+** _for instance:_ `topic:foo` changes to `:foo`
+** _for instance:_ `stream create stream1 --definition ":foo > log"` 
+
+==== Directed Graphs
+If you’re building non-linear streams, you could take advantage of named destinations to build 
+directed graphs.
+
+_for instance, in Spring XD:_
+
+[source,xml]
+----
+stream create f --definition "queue:foo > transform --expression=payload+'-foo' | log" --deploy
+stream create b --definition "queue:bar > transform --expression=payload+'-bar' | log" --deploy
+stream create r --definition "http | router --expression=payload.contains('a')?'queue:foo':'queue:bar'" --deploy
+----
+
+_for instance, in Spring Cloud Data Flow:_
+
+[source,xml]
+----
+stream create f --definition ":foo > transform --expression=payload+'-foo' | log" --deploy
+stream create b --definition ":bar > transform --expression=payload+'-bar' | log" --deploy
+stream create r --definition "http | router --expression=payload.contains('a')?':foo':':bar'" --deploy
+----
+
+=== Batch to Tasks
+
+A Task by definition, is any process that does not run forever, including Spring Batch jobs, and they 
+end/stop at some point. Task applications can be majorly used for on-demand use-cases such as database migration, machine learning, scheduled operations etc. Using link:http://cloud.spring.io/spring-cloud-task/[Spring Cloud Task], users can build Spring Batch jobs as microservice applications.
+
+* Spring Batch link:http://docs.spring.io/spring-xd/docs/current-SNAPSHOT/reference/html/#batch[jobs] 
```

Comment:
Would we rather use this link?
http://docs.spring.io/spring-xd/docs/current-SNAPSHOT/reference/html/#jobs